asusrouter/modules/endpoint/hook.py

```diff
@@ -733,3 +740,22 @@ def process_wlan(
         wlan[interface.value] = info
 
     return wlan
+
+def process_dsl(dsl_info: dict[str, Any]) -> dict[str, Any]:
+    """Process DSL data"""
+
+    dsl: dict[str, Any] = {}
+
+    # Data preprocessing
+    if "Kbps" in dsl_info.get("dsllog_dataratedown", ""):
+        dsl_info["dsllog_dataratedown"] = dsl_info["dsllog_dataratedown"].replace("Kbps", "")
+
+    if "Kbps" in dsl_info.get("dsllog_datarateup", ""):
+        dsl_info["dsllog_datarateup"] = dsl_info["dsllog_datarateup"].replace("Kbps", "")
+
+    dsl = {
+        "dsllog_dataratedown": safe_int(dsl_info["dsllog_dataratedown"], 0),
+        "dsllog_datarateup": safe_int(dsl_info["dsllog_datarateup"], 0),
+    }
+
+    return dsl
```

Comment:
```suggestion
def process_dsl(dsl_info: dict[str, Any]) -> dict[str, Any]:
    """Process DSL data"""

    def remove_units(value: Optional[str]) -> Optional[str]:
        """Remove units from the value."""

        if value is None:
            return None
        return value.split(" ")[0]

    dsl: dict[str, Any] = {}

    # Data preprocessing
    _dataratedown = remove_units(dsl_info.get("dsllog_dataratedown"))
    _datarateup = remove_units(dsl_info.get("dsllog_datarateup"))

    dsl["datarate"] = {
        "down": safe_int(_dataratedown, 0),
        "up": safe_int(_datarateup, 0),
    }

    return dsl

```

This will:

- reduce the number of dict lookups
- simplify the output

Reply:
Okay. I have update your suggestion into the code and tested it. It works.

![image](https://github.com/user-attachments/assets/83e8e49a-26b0-43a5-99cf-6094c1ba7c3c)

Thanks.